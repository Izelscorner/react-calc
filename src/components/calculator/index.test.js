import React from 'react';
import { mount } from 'enzyme';
import Index from './index';
import toJson from 'enzyme-to-json';
import ButtonList from './controls/buttonList'
import Display from './controls/display'

describe('<Index />', () => {
  let wrapper;

  beforeAll(() => {
    wrapper = mount(<Index />);
  });

  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('render button list ', () => {
    let buttonList = wrapper.find(ButtonList);
    expect(buttonList).toHaveLength(1);
  });

  it('render display ', () => {
    let display = wrapper.find(Display);
    expect(display).toHaveLength(1);
  });

  it('check initial state value', () => {
    const state = wrapper.state();
    expect(state.outcome).toEqual(null);
    expect(state.current).toEqual(null);
    expect(state.operator).toEqual(null);
  });

});
