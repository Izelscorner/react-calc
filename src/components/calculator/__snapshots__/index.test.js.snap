// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Index /> matches snapshot 1`] = `
<Calculator>
  <div
    className="calculator"
  >
    <Display
      display="0"
    >
      <div
        className="display"
      >
        <div
          className="display-branding"
        />
        <label>
          0
        </label>
      </div>
    </Display>
    <ButtonList
      handleEvent={[Function]}
    >
      <div
        className="button-list"
      >
        <div
          className="button-list__row"
        >
          <Button
            additionalClassName="button-size-75"
            handleEvent={[Function]}
            value="AC"
          >
            <button
              className="button button-size-75"
              onClick={[Function]}
            >
              AC
            </button>
          </Button>
          <Button
            additionalClassName="button--orange"
            handleEvent={[Function]}
            value="*"
          >
            <button
              className="button button--orange"
              onClick={[Function]}
            >
              *
            </button>
          </Button>
        </div>
        <div
          className="button-list__row"
        >
          <Button
            handleEvent={[Function]}
            value="7"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              7
            </button>
          </Button>
          <Button
            handleEvent={[Function]}
            value="8"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              8
            </button>
          </Button>
          <Button
            handleEvent={[Function]}
            value="9"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              9
            </button>
          </Button>
          <Button
            additionalClassName="button--orange"
            handleEvent={[Function]}
            value="-"
          >
            <button
              className="button button--orange"
              onClick={[Function]}
            >
              -
            </button>
          </Button>
        </div>
        <div
          className="button-list__row"
        >
          <Button
            handleEvent={[Function]}
            value="4"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              4
            </button>
          </Button>
          <Button
            handleEvent={[Function]}
            value="5"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              5
            </button>
          </Button>
          <Button
            handleEvent={[Function]}
            value="6"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              6
            </button>
          </Button>
          <Button
            additionalClassName="button--orange"
            handleEvent={[Function]}
            value="+"
          >
            <button
              className="button button--orange"
              onClick={[Function]}
            >
              +
            </button>
          </Button>
        </div>
        <div
          className="button-list__row"
        >
          <Button
            handleEvent={[Function]}
            value="1"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              1
            </button>
          </Button>
          <Button
            handleEvent={[Function]}
            value="2"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              2
            </button>
          </Button>
          <Button
            handleEvent={[Function]}
            value="3"
          >
            <button
              className="button"
              onClick={[Function]}
            >
              3
            </button>
          </Button>
          <Button
            additionalClassName="button--orange"
            handleEvent={[Function]}
            value="/"
          >
            <button
              className="button button--orange"
              onClick={[Function]}
            >
              /
            </button>
          </Button>
        </div>
        <div
          className="button-list__row"
        >
          <Button
            additionalClassName="button-size-50"
            handleEvent={[Function]}
            value="0"
          >
            <button
              className="button button-size-50"
              onClick={[Function]}
            >
              0
            </button>
          </Button>
          <Button
            handleEvent={[Function]}
            value="."
          >
            <button
              className="button"
              onClick={[Function]}
            >
              .
            </button>
          </Button>
          <Button
            additionalClassName="button--orange"
            handleEvent={[Function]}
            value="="
          >
            <button
              className="button button--orange"
              onClick={[Function]}
            >
              =
            </button>
          </Button>
        </div>
      </div>
    </ButtonList>
  </div>
</Calculator>
`;
